Course list/bl 97

pmAllport opened the pull request:
* Upgraded from bloc_test 7.0.0 -> 8.0.0
* Upgraded from flutter_bloc 6.0.0 -> 7.0.0
* Upgraded from bloc 6.0.0 -> 7.0.0
* Added mocktail 0.1.1 to dependencies (used by bloc-test)
* Fixed breaking errors of tests when upgrading from bloc_test 7.0.0 -> 8.0.0 (ref : https://pub.dev/packages/bloc_test/changelog) 
* Fixed "onError"-funtion in lib/blocs/simple_bloc_observer.dart

pmAllport commented:
Need to finish development before we push to master

amundlrohne commented on lib/blocs/simple_bloc_observer.dart:
```diff
@@ -14,8 +14,8 @@ class SimpleBlocObserver extends BlocObserver {
-  void onError(Cubit cubit, Object error, StackTrace stackTrace) {
-    print('onError $error');
-    super.onError(cubit, error, stackTrace);
+  void onError(BlocBase bloc, Object error, StackTrace stackTrace) {
```
Is this for better stacktraces?

pmAllport replied:
Should be. Its the one used in the bloc-documentation for BlocObservers : https://pub.dev/documentation/bloc/latest/

pmAllport replied:
It is also worth mentioning, that this crashes on the newest version of bloc_test (8.0.0). I'm guessing its due to bloc_test implementing a "MockCubit", which means that that function no longer can cast to Bloc, as they are now two seperate functions.

amundlrohne replied:
> A lot of spread out comments here.
> 
> -The obvious dev-code will be removed. This is being worked on as we speak
> -The tests have not yet been implemented. "CreateBlocProviderForTesting" is legacy and will be removed.
> 
> * The routing is also being worked on. Coming in its own PR.

Awesome, will approve it. Merge when you feel ready! :)

amundlrohne commented on lib/views/course/alternative.dart:
```diff
@@ -0,0 +1,58 @@
+          alignment: FractionalOffset.center,
+          decoration: widget.isPressed
+              ? BoxDecoration(
+                  borderRadius: BorderRadius.circular(5),
```
Only border radius if pressed?

amundlrohne commented on lib/views/course/alternative.dart:
```diff
@@ -0,0 +1,58 @@
+  Widget build(BuildContext context) {
+    return GestureDetector(
+        onTap: () {
+          debugPrint('triggerd');
```
remove

amundlrohne replied:
The debugPrint that is

amundlrohne commented on lib/views/course/alternative_container.dart:
```diff
@@ -0,0 +1,98 @@
+                setState(() => alts = alts.sublist(0, 1)),
+                debugPrint('CLICKCLICK: ' + alts.toString()),
+
+                /* BlocProvider.of<CourseDetailBloc>(context).add(
+                    CourseDetailRequested(
+                        course: widget.course,
+                        isAnswer: true,
+                        answerId: 1,
+                        isQuiz: true)) */
```
remove

amundlrohne commented on lib/views/course/alternative_container.dart:
```diff
@@ -0,0 +1,98 @@
+  }
+}
+
+//   @override
+//   Widget build(BuildContext context) {
+//     return Container(
+//         child: GridView.count(
+//             crossAxisCount: 2,
+//             childAspectRatio: 1,
+//             crossAxisSpacing: 20,
+//             mainAxisSpacing: 20,
+//             children: List.generate(list.length, (index) {
+//               return Alter.Alternative(
+//                   index, list[index]['name'], list[index]['correct'], null);
+//             })));
+//   }
+// }
```
remove

amundlrohne commented on lib/views/course/course.dart:
```diff
@@ -1,54 +1,104 @@
+    return BlocBuilder<CourseDetailBloc, CourseDetailState>(
+      // ignore: missing_return
+      builder: (context, state) {
+        //log(state.toString());
```
You guessed it

amundlrohne commented on lib/views/course/course.dart:
```diff
@@ -59,6 +109,15 @@ class _CardContainerViewState extends State<CardContainerView> {
+          indx = (_controller.offset / context.size.width).roundToDouble();
+        });
+      }
+      print(indx);
```
remove

amundlrohne commented on test/utils/bloc/mock_bloc.dart:
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+Widget createBlocProviderForTesting({
+  Widget child,
+  Bloc bloc,
+}) {
+  return MaterialApp(
+    home: BlocProvider(
+      create: (context) => bloc,
+      child: child,
+    ),
+  );
+}
```
Where is this used?

amundlrohne reviewed (commented):
Implementation looks good. That being said, there are some things that should be cleaned up. Mostly commented out code, and debug prints. 

Also, will routing from course_list to course_detail be in a seperate PR?

pmAllport commented:
A lot of spread out comments here.

-The obvious dev-code will be removed. This is being worked on as we speak
-The tests have not yet been implemented. "CreateBlocProviderForTesting" is legacy and will be removed.
- The routing is also being worked on. Coming in its own PR.

amundlrohne reviewed (approved):
Approved!

pmAllport commented:
* Pulled from master and resolved mergeconflicts
* Added some courseDetail tests (Event, State and Bloc).
* Still missing Course tests
* Fixed layout in slides
* Added button to end of slides to lead to "quiz"

pmAllport commented:
Fixed flutter analyze issues

amundlrohne commented on lib/blocs/course_detail/course_detail_bloc.dart:
```diff
@@ -0,0 +1,91 @@
+        return;
+      }
+    }
+    yield Failed(err: 'Error, bad request');
```
Good job with these Failed states. Would be nice with some clearer messages though.

amundlrohne commented on lib/views/course/alternative_container.dart:
```diff
@@ -0,0 +1,91 @@
+      Container(
+          alignment: Alignment.topCenter,
+          padding: EdgeInsets.symmetric(vertical: 20),
+          child: Text(validation, style: TextStyle(color: Colors.teal))),
```
Could be improved by retrieving textstyle from themes

amundlrohne reviewed (approved):
Looks good to me! A tad big, but all of the improvements in this PR should prevent this in the future. 👍
